worker/uniter/leadership/resolver.go

```diff
@@ -32,7 +32,7 @@ func (l *leadershipResolver) NextOp(
 ) (operation.Operation, error) {
 
 	// TODO(wallyworld) - maybe this can occur before install
-	if !opState.Installed {
+	if !opState.Installed || opState.Kind == operation.RunHook {
```

Comment:
You're right that we shouldn't do *most* of the leadership stuff during hook error, but there's a bit in ModeHookError that deals with resigning leadership (only)